plugins/leemons-plugin-academic-portfolio/frontend/src/pages/private/AcademicTreePage.js

```diff
@@ -0,0 +1,314 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
+import { useUserCenters } from '@users/hooks';
+import {
+  Select,
+  TotalLayoutContainer,
+  TotalLayoutHeader,
+  Stack,
+  ImageLoader,
+  Box,
+} from '@bubbles-ui/components';
+import useProgramAcademicTree from '@academic-portfolio/hooks/queries/useProgramAcademicTree';
+import useProgramsByCenter from '@academic-portfolio/hooks/queries/useCenterPrograms';
+import EnrollmentDrawer from '@academic-portfolio/components/AcademicTree/EnrollmentDrawer/EnrollmentDrawer';
+
+import useTranslateLoader from '@multilanguage/useTranslateLoader';
+import prefixPN from '@academic-portfolio/helpers/prefixPN';
+import { CourseView } from '@academic-portfolio/components/AcademicTree/CourseView/CourseView';
+import { KnowledgeView } from '@academic-portfolio/components/AcademicTree/KnowledgeView/KnowledgeView';
+import TreeBox from '@academic-portfolio/components/AcademicTree/TreeBox';
+import { GroupView } from '../../components/AcademicTree/GroupView/GroupView';
+import SubjectView from '../../components/AcademicTree/SubjectView/SubjectView';
+
+const AcademicTreePage = () => {
+  const [selectedCenter, setSelectedCenter] = useState('');
+  const [selectedProgram, setSelectedProgram] = useState('');
+  const [selectedTreeNode, setSelectedTreeNode] = useState(null);
+  const [enrollmentDrawerIsOpen, setEnrollmentDrawerIsOpen] = useState(false);
+
+  const [t] = useTranslateLoader(prefixPN('tree_page'));
+  const [enrolllmentDrawerOpenedFromClassroom, setEnrolllmentDrawerOpenedFromClassroom] =
+    useState(null);
+  const { data: userCenters, isLoading: areCentersLoading } = useUserCenters();
+  const scrollRef = useRef();
+  const history = useHistory();
+  const viewRef = useRef();
+
+  const handleNodeClick = (nodeId) => {
+    setSelectedTreeNode(nodeId);
+  };
+
+  // SET UP ------------------------------------------------------------------------------------------------ ||
+  const centersData = useMemo(
+    () => userCenters?.map((center) => ({ value: center?.id, label: center?.name })),
+    [userCenters]
+  );
+
+  const { data: centerProgramsQuery, isLoading: areCenterProgramsLoading } = useProgramsByCenter({
+    center: selectedCenter,
+    filters: { onlyActive: true },
+    options: {
+      enabled: selectedCenter?.length > 0,
+    },
+  });
+
+  const programSelectData = useMemo(() => {
+    if (centerProgramsQuery?.length) {
+      return [
+        ...centerProgramsQuery.map((program) => ({ value: program.id, label: program.name })),
+      ];
+    }
+    return [];
+  }, [centerProgramsQuery]);
+
+  const { data: academicTreeQuery, isLoading: isAcademicTreeLoading } = useProgramAcademicTree({
+    programId: selectedProgram,
+    options: { enabled: selectedProgram?.length > 0 },
+  });
+
+  const generateGUID = () => `_${Math.random().toString(36).substr(2, 9)}`;
```

Comment:
Watch out, this is not unique, I recommend to use `uuid`

Reply:
Hecho!